modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/BuildResultPayload.java

```diff
@@ -17,119 +17,37 @@
  */
 package org.wso2.mdm.agent.services;
 
-import org.json.JSONArray;
-import org.json.JSONException;
-import org.json.JSONObject;
-import org.wso2.mdm.agent.R;
+import org.wso2.mdm.agent.beans.Operation;
 
 import android.content.Context;
-import android.content.res.Resources;
-import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.List;
 
 /**
  * This class handles building of the operation list result payload to be sent to the server.
  */
 public class BuildResultPayload {
-	private JSONArray operationListResult;
-	private JSONObject result;
-	private Resources resources;
-	private String TAG = BuildResultPayload.class.getName();
-	
-	public BuildResultPayload(Context context){
-		this.resources = context.getResources();
-		this.operationListResult = new JSONArray();
-		this.result = new JSONObject();
-	}
-	
-	/**
-	 * Add operation results to the array to be returned.
-	 * @param code     - Operation code.
-	 */
-	public void build(String code) {
-		try {
-			result.put(resources.getString(R.string.intent_extra_status),
-				           resources.getString(R.string.shared_pref_default_status));
-			
-			result.put(resources.getString(R.string.intent_extra_code), code);
-			
-			operationListResult.put(result);
-		} catch (JSONException e) {
-			Log.e(TAG, "Invalid JSON format." + e.toString());
-		}
-	}
-	
-	/**
-	 * Add operation results to the array to be returned.
-	 * @param code     - Operation code.
-	 * @param data     - Data JSON object.
-	 */
-	public void build(String code, JSONObject data) {
-		try {
-			result.put(resources.getString(R.string.intent_extra_status),
-				          resources.getString(R.string.shared_pref_default_status));
 
-			result.put(resources.getString(R.string.intent_extra_code), code);
-			
-			if (data != null) {
-				result.put(resources.getString(R.string.intent_extra_data), data);
-			}
-			
-			operationListResult.put(result);
-		} catch (JSONException e) {
-			Log.e(TAG, "Invalid JSON format." + e.toString());
-		}
-	}
-	
-	/**
-	 * Add operation results to the array to be returned.
-	 * @param code     - Operation code.
-	 * @param dataList - Data JSON Array.
-	 */
-	public void build(String code, JSONArray dataList) {
-		try {
-			result.put(resources.getString(R.string.intent_extra_status),
-			           	resources.getString(R.string.shared_pref_default_status));
-			
-			result.put(resources.getString(R.string.intent_extra_code), code);
+	private List<Operation> operationResponses;
 
-			if (dataList != null) {
-				result.put(resources.getString(R.string.intent_extra_data), dataList);
-			}
-			
-			operationListResult.put(result);
-		} catch (JSONException e) {
-			Log.e(TAG, "Invalid JSON format." + e.toString());
-		}
+	public BuildResultPayload(){
```

Comment:
Fixed in Commit [https://github.com/wso2/product-mdm/commit/b0c66506d24fb5288300bec4224f1eaa58c7b426]